app-header-layout/app-header-layout.html

```diff
@@ -155,6 +163,48 @@
         }, 3);
       },
 
+      // When app-header-layout is disabled, we disable scrolling on the main content. On non-iOS
+      // devices, we do this by setting <body>.style.overflow = 'hidden'; this is done in
+      // <app-drawer> so that the functionality is provided when not using <app-header-layout>.
+      // This does not work on iOS devices which does scrolling on a separate layer; to overcome
+      // this, we use hacks that require a certain DOM structure, so this functionality is only
+      // available with <app-header-layout>.
+      _disabledChanged: function(disabled) {
+        var header = this.header;
+        if (!header) {
+          return;
+        }
+
+        if (disabled) {
+          if (this.hasScrollingRegion) {
+            // If scrolling is done inside an element with webkitOverflowScrolling = 'touch' (i.e.
+            // not document scrolling), then we can just set webkitOverflowScrolling = 'auto' to
+            // disable scrolling.
+            this.$.contentContainer.style.webkitOverflowScrolling = 'auto';
+          } else {
+            // If we are using document scrolling, we disable the header to stop any scrolling
+            // effects, resize the height <app-header-layout> to the viewport height, and
+            // vertically translate the contents to simulate disabled scrolling. We also save the
+            // last scroll position so we can restore it later.
+            header.disabled = true;
+            this._savedPageYOffset = window.pageYOffset;
+            this.style.height  = window.innerHeight + 'px';
+            this.style.overflow = 'hidden';
+            this.transform('translateY(-' + this._savedPageYOffset + 'px)', this.$.contentContainer);
+          }
+        } else {
+          if (this.hasScrollingRegion) {
+            this.$.contentContainer.style.webkitOverflowScrolling = '';
+          } else {
+            this.style.height = '';
+            this.style.overflow = '';
+            this.transform('', this.$.contentContainer);
+            window.scrollTo(window.pageXOffset, this._savedPageYOffset);
```

Comment:
@blasten you are right.  this will reset the scroll pos which is probably not what we want.